### PR TITLE
feat(project): improve build and configurations

### PR DESCRIPTION
**Pull request by e-cloud**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [x] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
some non-up-to-date deps and not good enough tools configurations.

## What is the new behavior?
improvement over the problem. updated deps and clearer configurations.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -17,12 +18,17 @@ module.exports = function(config) {
...
+      reports: ['html', 'lcovonly'],
       fixWebpackSourcePaths: true
     },
+    angularCli: {
+      environment: 'production'
+    },
     reporters: ['progress', 'kjhtml'],
     port: 9876,
     colors: true,
+    logLevel: config.LOG_INFO,
```
Comment (splincode) [inline karma.conf.js]: why you need usage logLevel instead?

```ts
browserConsoleLogOptions: {
  level: CI ? 'error' : 'debug'
}
```

**Comment by splincode** [comment]
You should not mix changes in one branch.

File: docs/concepts/select.md [hunk 2]
```diff
@@ -264,9 +264,9 @@ export class ZooState {
 }
 ```
 
-When using the `Selector` decorator along with a state class, it will still
-inject the state class's state first followed by the other selectors in the order
-they were passed in the signature.
+> NOTE: When using the `Selector` decorator along with a state class, **it will still
```
Comment (splincode) [inline docs/concepts/select.md]: Take out in a separate PR

File: tsconfig.build.json [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
+    "lib": [
+      "dom",
+      "es2017"
+    ],
+    "paths": {
+      "@ngxs/*": ["@ngxs/*"]
+    },
+  },
+  "angularCompilerOptions": {
+    "annotateForClosureCompiler": true,
```
Comment (splincode) [inline tsconfig.build.json]: What are the advantages?

File: tsconfig.lib.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    "sourceMap": true,
+    "inlineSources": true,
+    "importHelpers": true,
+    "types": [],
+    "lib": [
+      "dom",
+      "es2017"
+    ]
+  },
+  "angularCompilerOptions": {
```
Comment (splincode) [inline tsconfig.lib.json]: Why not immediately take out a separate file `tsconfig.json` and not inherit from it?

File: karma.conf.js (hunk 1 above)
Comment (e-cloud) [reply karma.conf.js]: @splincode these two are with different scope. loglevel is for karma internal log, browserConsoleLogOptions is for reporter.

File: tsconfig.build.json (hunk 3 above)
Comment (e-cloud) [reply tsconfig.build.json]: this will generate some information in output, which could be used by user using closure.

File: karma.conf.js (hunk 1 above)
Comment (splincode) [reply karma.conf.js]: but we will be moving to jest soon and karma internal log was not really needed

File: tsconfig.build.json (hunk 3 above)
Comment (splincode) [reply tsconfig.build.json]: Does it increase the bundle?

Comment (splincode) [reply tsconfig.build.json]: good

**Comment by e-cloud** [comment]
@splincode could you review again?

File: tsconfig.lib.json (hunk 4 above)
Comment (splincode) [reply tsconfig.lib.json]: angularCompilerOptions move to tsconfig.json

**Comment by splincode** [comment]
@markwhitfeld could you take a look?

**Comment by e-cloud** [comment]
@splincode @markwhitfeld any progress?

**Comment by e-cloud** [comment]
@markwhitfeld please have a review

**Comment by markwhitfeld** [comment]
@e-cloud I'm happy if the others are happy, just testing locally before I merge...

**Pull request merged**